src/NuGet.Clients/NuGet.VisualStudio.Common/Telemetry/NuGetPowerShellUsageCollector.cs

```diff
@@ -0,0 +1,365 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using NuGet.Common;
+using NuGet.VisualStudio.Telemetry.PowerShell;
+
+namespace NuGet.VisualStudio.Telemetry
+{
+    public sealed class NuGetPowerShellUsageCollector : IDisposable
+    {
+        // PMC, PMUI powershell telemetry consts
+        public const string PmcExecuteCommandCount = nameof(PmcExecuteCommandCount);
+        public const string PmuiExecuteCommandCount = nameof(PmuiExecuteCommandCount);
+        public const string NuGetCommandUsed = nameof(NuGetCommandUsed);
+        public const string InitPs1LoadPmui = nameof(InitPs1LoadPmui);
+        public const string InitPs1LoadPmc = nameof(InitPs1LoadPmc);
+        public const string InitPs1LoadedFromPmcFirst = nameof(InitPs1LoadedFromPmcFirst);
+        public const string LoadedFromPmui = nameof(LoadedFromPmui);
+        public const string FirstTimeLoadedFromPmui = nameof(FirstTimeLoadedFromPmui);
+        public const string LoadedFromPmc = nameof(LoadedFromPmc);
+        public const string FirstTimeLoadedFromPmc = nameof(FirstTimeLoadedFromPmc);
```

Comment:
I think there was some equivalent feedback through the PR...instead of `FirstTimeLoadedFromPmc`, consider `FirstTimeLoadReason` and adding pmc and pm ui as options.

Reply:
We already have below one for that purpose. But this is one is for if Pmc host loaded first or not in current solution session.  If it was already loaded during previous solution session then it would be `false` even though PMC is active now.

https://github.com/NuGet/NuGet.Client/blob/7c710c0970cbe2c6935e0ae039a4b6f543cbf1b1/src/NuGet.Clients/NuGetConsole.Host.PowerShell/PowerShellHost.cs#L322